### PR TITLE
feat: [GM-52] [BE] Implement the update package endpoint

### PR DESCRIPTION
**Pull request by krishankumar01**

### CODE REVIEW

File: src/go/package/app/grpc/controllers/validation.go
Comment (hwbrzzl) [inline src/go/package/app/grpc/controllers/validation.go:1]: We can add a helper method for validateUpdatePackageRequest and validateCreatePackageRequest, they have some same validators.

File: src/go/package/app/models/package_test.go [hunk 1]
```diff
@@ -29,6 +29,168 @@ func (s *PackageSuite) SetupTest() {
...
+				mockOrmQuery.On("Association", "Tags").Return(mockOrmAssociation).Once()
+				mockOrmAssociation.On("Replace", mock.MatchedBy(func(tags []*Tag) bool {
+					return len(tags) == 1 && tags[0].Name == "goravel"
+				})).Return(nil).Once()
+			},
+		},
+		{
+			name: "Sad path - Create Tag error",
+			setup: func() {
+				beforeSetup()
```
Comment (hwbrzzl) [inline src/go/package/app/models/package_test.go]: Why add this here? It exists in the for logic below.

File: src/go/package/app/models/package_test.go [hunk 2]
```diff
@@ -149,3 +311,71 @@ func (s *PackageSuite) TestToProto() {
...
+				mockOrmQuery.On("Save", mock.MatchedBy(func(pkg *Package) bool {
+					return pkg.Name == name && pkg.Link == url && pkg.UserID == userID
+				})).Return(nil).Once()
+			},
+			expectedErr: nil,
+		},
+		{
+			name: "Sad path - save package error",
+			setup: func() {
+				beforeSetup()
```
Comment (hwbrzzl) [inline src/go/package/app/models/package_test.go]: Ditto

File: src/go/package/app/services/package.go
Comment (hwbrzzl) [inline src/go/package/app/services/package.go:1]: Great

File: src/go/package/app/services/package_test.go [hunk 3]
```diff
@@ -562,3 +566,159 @@ func (s *PackageTestSuite) TestGetPackages() {
...
+		s.Run(test.name, func() {
+			test.setup()
+
+			pkg, err := s.packageImpl.UpdatePackage(s.ctx, test.request)
+			if test.expectedErr != nil {
+				s.Nil(pkg)
+				s.Equal(test.expectedErr, err)
+			} else {
+				s.Nil(err)
+				s.Equal(test.expectPackage, pkg)
```
Comment (hwbrzzl) [inline src/go/package/app/services/package_test.go:720]: Missing `AssertExpectations`

File: src/go/package/app/models/package_test.go (hunk 1 above)
Comment (krishankumar01) [reply src/go/package/app/models/package_test.go]: sorry by mistake

**Review by hwbrzzl — APPROVED** [review]
Beautiful

**Pull request merged**